robotoff/app/core.py

```diff
@@ -107,5 +138,56 @@ def save_insight(
     if insight.annotation is not None:
         return ALREADY_ANNOTATED_RESULT
 
+    if verify_annotation:
+        verified: bool = False
+
+        AnnotationVote.create(
+            insight_id=insight_id,
+            username=auth.get_username() if auth else None,
+            value=annotation,
+            device_id=device_id,
+        )
+
+        with db.atomic() as tx:
```

Comment:
why don't we run all the save_annotation function atomically ? (that would make more sense).

Reply:
in fact in annotator.annotate we will open a distinct transaction. But doesn't that mean, we can be in an inconsistent state ?

Reply:
The code would be more complex if we wanted all of it in a transaction, and transactions are expensive.

At the same time, the only thing important to keep atomic there is that the n_votes is correct, so that insights don't stay at the bottom of the pile by mistake.

If two votes arrive at the same time, the worst thing that could happen currently is that the second vote would override the first vote, but I'm not sure that is bad enough to warrant extra code complexity.

WDYT?

Reply:
I'm ok with it.
As you say there no really evil consequences.